app/packages/viz3d/src/point_cloud_looker.ts

```diff
@@ -0,0 +1,321 @@
+    mediaBBox: [0, 0, 0, 0],
+    transformedMediaBBox: [0, 0, 0, 0],
+    canvasBBox: [0, 0, 0, 0],
+    textPad: 0,
+    pointRadius: 0,
+    dashLength: 0,
+    relativeCoordinates: [0, 0],
+    mouseIsOnOverlay: false,
+    showHelp: false,
+    overlaysPrepared: false,
+    disableOverlays: false,
+    zoomToContent: false,
+    setZoom: false,
+    hasDefaultZoom: false,
+    SHORTCUTS: undefined,
+    error: 0,
+    destroyed: false,
+    reloading: false
+};
+
+export class PointCloudSample implements Sample {
+    metadata: { width: number; height: number; };
+    id: string = "";
+    media_type: "model" = "model";
+    filepath: string = "";
+    compressed_path: string = "";
+    tags: string[] = [];
+    _label_tags: string[] = [];
+
+};
+
+export interface PointCloudOptions extends BaseOptions {
+    zoom: boolean;
+};
+
+export const DEFAULT_POINTCLOUD_OPTIONS = {
+    ...DEFAULT_BASE_OPTIONS,
+    zoom: false
+}
+
+export class PointCloudElement extends BaseElement<PointCloudState, HTMLCanvasElement>{
+    private _canvas: HTMLCanvasElement;
+    private _ctx: CanvasRenderingContext2D;
+    private _display: pcd.Display3D<HTMLCanvasElement>;
+    private _thumb_gen: ThumbnailGenerator;
+    private _thumb_image: HTMLImageElement;
+
+    public createHTMLElement(update: StateUpdate<PointCloudState>, dispatchEvent: (eventType: string, details?: any) => void): HTMLCanvasElement {
+        this._canvas = document.createElement("canvas");
+        this._canvas.width = window.innerWidth;
+        this._canvas.height = window.innerHeight;
+        this._ctx = this._canvas.getContext("2d");
+        this._thumb_image = new Image();
+
+        update((state: Readonly<PointCloudState>) => {
+            this._display = create3DDisplay(state.config.displayConfig);
+            this._thumb_gen = _global_thumbnail_generator.make(state.config.thumbnail_width, state.config.thumbnail_height, state.config.displayConfig);
+            return state;
+        });
+
+        return this._canvas;
+    }
+
+    private _renderThumbnail (mesh: three.Mesh) {
+        _global_render_loop.stop();
+        this._thumb_gen.makeThumbnailURL(mesh).then((url) => {
+            this._thumb_image.src = url;
+            this._ctx.drawImage(this._thumb_image, 0, 0);
+        });
+    }
+
+    private _renderScene (mesh: three.Mesh) {
+        _global_render_loop.stop();
+
+        this._display.clearScene();
+        this._display.addSceneItem(mesh);
+        this._display.setInteractionElement(this._canvas);
+        this._display.setSize(this._canvas.width, this._canvas.height);
+
+        _global_render_loop.start(() => {
+            this._display.render();
+            this._display.copyToCanvas(this._ctx);
+        })
+    }
+
+    public renderSelf(state: Readonly<PointCloudState>, sample: Readonly<Sample>): HTMLCanvasElement {
+        _global_draco_loader.then((loader) => {
+            // TODO: Need to cache results here... 
```

Comment:
Is there any sort of higher level caching layer I should be using? Any react state trickiness I need to be considering here?

Reply:
I added a rudimentary caching layer. Let me know if I should do something differently.